vim-patch:8.1.1483: skipped tests are not properly listed

blueyed opened the pull request:
Problem:    Skipped tests are not properly listed.
Solution:   Throw a "Skipped" exception instead of using ":finish" or ":return".
https://github.com/vim/vim/commit/5d30ff19648d2ff0696cea97582b902f6a4ec0ba

Skips quite some (N/A) tests.

vim-patch:8.1.0503: missing change to diff test (included in cf1ffa916)

blueyed committed:
vim-patch:8.1.1483: skipped tests are not properly listed

Problem:    Skipped tests are not properly listed.
Solution:   Throw a "Skipped" exception instead of using ":finish" or ":return".
https://github.com/vim/vim/commit/5d30ff19648d2ff0696cea97582b902f6a4ec0ba

Skips quite some (N/A) tests.

vim-patch:8.1.0503: missing change to diff test (included in cf1ffa916)

blueyed commented:
Missing the patch to handle "Skipped" still?

> Caught exception: Skipped: job and/or clientserver feature missing @ 
https://travis-ci.org/neovim/neovim/jobs/574115954#L8362

blueyed committed:
vim-patch:8.1.0811: too many #ifdefs

Problem:    Too many #ifdefs.
Solution:   Graduate FEAT_MBYTE, the final chapter.
https://github.com/vim/vim/commit/30276f2beb248557c6b33cd5418bca8b7084b0a5

blueyed committed:
vim-patch:8.1.1476: no statistics displayed after running tests

Problem:    No statistics displayed after running tests.
Solution:   Summarize the test results. (Christian Brabandt, closes vim/vim#4391)
            Also make it possible to report a skipped file.
https://github.com/vim/vim/commit/9c0cec65f891492314caadeef87a50251a21e630

Removes our custom no-inits from `$(RUN_VIMTEST)`, since we have
`$(NO_INITS)` now also.

blueyed commented:
Summary for oldtest \o/ 

Travis:

```
[OLDTEST] Running test_alot
../../../build/bin/nvim -u unix.vim -U NONE --noplugin --headless -S summarize.vim messages
E484: Can't open file /home/travis/nvim-install/share/nvim/syntax/syntax.vim81 matches on 81 lines
48 matches on 48 lines
Test results:
Skipped:
	Test_BufLeave_Wipe(): TODO: 
	Test_ChangedP(): see test/functional/viml/completion_spec.lua
	Test_OptionSet(): Nvim does not support test_override()
	Test_OptionSet_diffmode(): Nvim does not support test_override()
	Test_OptionSet_diffmode_close(): Nvim does not support test_override()
	Test_OptionSet_modeline(): Nvim does not support test_override()
	Test_TextChangedI_with_setline(): Nvim does not support test_override()
	test_clientserver.vim: job and/or clientserver feature missing
	Test_Debugger(): cannot run Vim in a terminal window
	Test_diff_of_diff(): cannot run Vim in a terminal window
	Test_diff_with_cursorline(): cannot run Vim in a terminal window
	Test_edit_08(): moved to test/functional/legacy/edit_spec.lua
	Test_edit_CTRL_R(): Nvim does not support test_override()
	Test_edit_CTRL_V(): Nvim does not support test_override()
	Test_folds_with_rnu(): cannot make screendumps
	Test_cursorline_after_yank(): cannot make screendumps
	Test_cursorline_with_visualmode(): cannot make screendumps
	Test_normal35_g_cmd4(): output of g< can't be tested currently
	Test_normal47_autocmd(): not possible to test cursorhold autocmd while waiting for input in normal_cmd
	Test_incsearch_substitute_dump(): cannot make screendumps
	Test_incsearch_with_change(): cannot make screendumps and/or timers feature and/or incsearch option missing
	Test_search_cmdline(): Nvim does not support test_override()
	Test_search_cmdline2(): Nvim does not support test_override()
	Test_search_cmdline3(): Nvim does not support test_override()
	Test_search_cmdline4(): Nvim does not support test_override()
	Test_sign_memfailures(): Nvim does not support test_alloc_fail()
	Test_spellinfo(): Nvim does not support enc=latin1
	Test_zz_NOSLITSUGS(): Nvim does not support enc=latin1
	Test_zz_Numbers(): Nvim does not support enc=latin1
	Test_zz_affix(): Nvim does not support enc=latin1
	Test_zz_basic(): Nvim does not support enc=latin1
	Test_zz_compound(): Nvim does not support enc=latin1
	Test_zz_prefixes(): Nvim does not support enc=latin1
	Test_zz_sal_and_addition(): Nvim does not support enc=latin1
	Test_compatible_args(): Nvim is always 'nocompatible'
	Test_startup_script(): Nvim does not need defaults.vim
	Test_sub_cmd_6(): Nvim removed POSIX-related 'cpoptions' flags
	Test_swapfile_delete(): need the "blob" feature for this test
	Test_syntax_c(): cannot make screendumps
	Test_peek_and_get_char(): Nvim does not support test_feedinput()
	Test_undo_del_chars(): Nvim does not support test_settime()
	Test_undofile_earlier(): Nvim does not support test_settime()
	Test_virtual_replace(): TODO: 
	Test_detect_ambiwidth(): cannot run Vim in a terminal window
	Test_isnan(): Nvim does not support isnan()
	Test_tabpage_cmdheight(): cannot make screendumps
	Test_tag_file_encoding(): Nvim removed test83-tags2, test83-tags3
	Test_translate_menu(): translated menu not found
-------------------------------
Executed:  1341 Tests
 Skipped:    48 Tests
  Failed:     0 Tests
ALL DONE
```

AppVeyor:
```
../../../build/bin/nvim.exe -u unix.vim -U NONE --noplugin --headless -S summarize.vim messages
E484: Can't open file C:/Program Files (x86)/nvim/share/nvim\syntax\syntax.vim80 matches on 80 lines
49 matches on 49 lines
1 match on 1 line
Test results:
Skipped:
	Test_BufLeave_Wipe(): TODO: 
	Test_ChangedP(): see test/functional/viml/completion_spec.lua
	Test_OptionSet(): Nvim does not support test_override()
	Test_OptionSet_diffmode(): Nvim does not support test_override()
	Test_OptionSet_diffmode_close(): Nvim does not support test_override()
	Test_OptionSet_modeline(): Nvim does not support test_override()
	Test_TextChangedI_with_setline(): Nvim does not support test_override()
	test_clientserver.vim: job and/or clientserver feature missing
	Test_Debugger(): cannot run Vim in a terminal window
	Test_diff_of_diff(): cannot run Vim in a terminal window
	Test_diff_with_cursorline(): cannot run Vim in a terminal window
	Test_edit_08(): moved to test/functional/legacy/edit_spec.lua
	Test_edit_CTRL_R(): Nvim does not support test_override()
	Test_edit_CTRL_V(): Nvim does not support test_override()
	Test_folds_with_rnu(): cannot make screendumps
	Test_cursorline_after_yank(): cannot make screendumps
	Test_cursorline_with_visualmode(): cannot make screendumps
	Test_normal35_g_cmd4(): output of g< can't be tested currently
	Test_normal47_autocmd(): not possible to test cursorhold autocmd while waiting for input in normal_cmd
	Test_quotestar(): Test is not implemented yet for this platform.
	Test_incsearch_substitute_dump(): cannot make screendumps
	Test_incsearch_with_change(): cannot make screendumps and/or timers feature and/or incsearch option missing
	Test_search_cmdline(): Nvim does not support test_override()
	Test_search_cmdline2(): Nvim does not support test_override()
	Test_search_cmdline3(): Nvim does not support test_override()
	Test_search_cmdline4(): Nvim does not support test_override()
	Test_sign_memfailures(): Nvim does not support test_alloc_fail()
	Test_spellinfo(): Nvim does not support enc=latin1
	Test_zz_NOSLITSUGS(): Nvim does not support enc=latin1
	Test_zz_Numbers(): Nvim does not support enc=latin1
	Test_zz_affix(): Nvim does not support enc=latin1
	Test_zz_basic(): Nvim does not support enc=latin1
	Test_zz_compound(): Nvim does not support enc=latin1
	Test_zz_prefixes(): Nvim does not support enc=latin1
	Test_zz_sal_and_addition(): Nvim does not support enc=latin1
	Test_compatible_args(): Nvim is always 'nocompatible'
	Test_startup_script(): Nvim does not need defaults.vim
	Test_sub_cmd_6(): Nvim removed POSIX-related 'cpoptions' flags
	Test_swapfile_delete(): need the "blob" feature for this test
	Test_syntax_c(): cannot make screendumps
	Test_peek_and_get_char(): Nvim does not support test_feedinput()
	Test_undo_del_chars(): Nvim does not support test_settime()
	Test_undofile_earlier(): Nvim does not support test_settime()
	Test_virtual_replace(): TODO: 
	Test_detect_ambiwidth(): cannot run Vim in a terminal window
	Test_isnan(): Nvim does not support isnan()
	Test_tabpage_cmdheight(): cannot make screendumps
	Test_tag_file_encoding(): Nvim removed test83-tags2, test83-tags3
	Test_translate_menu(): translated menu not found
-------------------------------
Executed:  1337 Tests
 Skipped:    49 Tests
  FAILED:     1 Tests
Failures: 
	From test_marks.vim:
	Found errors in Test_marks_cmd():
	function RunTheTest[37]..Test_marks_cmd line 12: Expected 9 but got 11
	function RunTheTest[37]..Test_marks_cmd line 18: Expected ' "      1    0 aaa' but got ' 0      1    0 C:/projects/neovim/src/nvim/testdir/dotest.in'
	function RunTheTest[37]..Test_marks_cmd line 19: Expected ' [      1    0 aaa' but got ' 1      1    0 C:/projects/neovim/src/nvim/testdir/test64.ok'
	function RunTheTest[37]..Test_marks_cmd line 20: Expected ' ]      2    0 bbb' but got ' "      1    0 aaa'
	function RunTheTest[37]..Test_marks_cmd line 21: Expected ' .      2    0 bbb' but got ' [      1    0 aaa'
	function RunTheTest[37]..Test_marks_cmd line 25: Expected 9 but got 11
	function RunTheTest[37]..Test_marks_cmd line 31: Expected ' "      2    0 ddd' but got ' 0      1    0 C:/projects/neovim/src/nvim/testdir/dotest.in'
	function RunTheTest[37]..Test_marks_cmd line 32: Expected ' [      1    0 ccc' but got ' 1      1    0 C:/projects/neovim/src/nvim/testdir/test64.ok'
	function RunTheTest[37]..Test_marks_cmd line 33: Expected ' ]      2    0 ddd' but got ' "      2    0 ddd'
	function RunTheTest[37]..Test_marks_cmd line 34: Expected ' .      2    0 ddd' but got ' [      1    0 ccc'
TEST FAILURE
```

blueyed committed:
vim-patch:8.1.1488: summary of tests has incorrect failed count

Problem:    Summary of tests has incorrect failed count.
Solution:   Add to the failed count instead of setting it. (Christian Brabandt)
https://github.com/vim/vim/commit/150f0550f45b836200a189e4d34417f4d3467455

blueyed committed:
vim-patch:8.1.1477: test summary fails in the tiny version

Problem:    Test summary fails in the tiny version.
Solution:   set 'nocompatible'.
https://github.com/vim/vim/commit/c7500f9cb5b9fbedb9f1a385fdc3790335054bd3

blueyed committed:
vim-patch:8.1.1478: still an error when running tests with the tiny version

Problem:    Still an error when running tests with the tiny version.
Solution:   Do not try reading test.log
https://github.com/vim/vim/commit/18250e291a5b4b8e32229d99ccb7c895ff0ca637

blueyed commented:
Not sure about this, likely fixed with later patches (added):

> E484: Can't open file /home/travis/nvim-install/share/nvim/syntax/syntax.vim81 matches on 81 lines
> 48 matches on 48 lines

justinmk commented on src/nvim/testdir/Makefile:
```diff
@@ -160,7 +160,7 @@ nolog:
 # to write and a lot easier to read and debug.
 # Limitation: Only works with the +eval feature.
-RUN_VIMTEST = $(TOOL) $(NVIM_PRG) -u unix.vim -U NONE -i viminfo --headless --noplugin
+RUN_VIMTEST = $(TOOL) $(NVIM_PRG) -u unix.vim
```
removing `--headless` seems risky. That's for Nvim after ZyX improved the output.

blueyed replied:
It is included via `$(NO_INITS)` now.

justinmk replied:
where is `NO_INITS` defined?

justinmk commented:
> Not sure about this, likely fixed with later patches (added):
> 
> > E484: Can't open file /home/travis/nvim-install/share/nvim/syntax/syntax.vim81 matches on 81 lines

still seeing that https://builds.sr.ht/~jmk/job/82517#bottom .

blueyed commented:
@justinmk 
This is due to `:s` commands used there, which Neovim displays then in headless mode: https://github.com/vim/vim/pull/4845

blueyed committed:
vim-patch:8.1.1893: script to summarize test results can be improved

Problem:    Script to summarize test results can be improved.
Solution:   Use "silent" for substitute to avoid reporting number of matches.
            Remove duplicate "set nocp". (Daniel Hahler, closes vim/vim#4845)
https://github.com/vim/vim/commit/f1e0544d413b8b2ea129db0655b4e123d54177a6

justinmk commented:
The "matches" messages are gone 🎉  But this remains:

    E484: Can't open file /usr/local/share/nvim/syntax/syntax.vim

https://builds.sr.ht/~jmk/job/82662#task-test-126

blueyed commented:
> ```
> E484: Can't open file /usr/local/share/nvim/syntax/syntax.vim
> ```

Likely because it's run before installation.

On Travis:

> E484: Can't open file /home/travis/nvim-install/share/nvim/syntax/syntax.vim

erw7 commented:
@blueyed The reason test fails on AppVeyor is because you stopped using the `-i` command line option. The `fixff` of #10679 also broke the user's `viminfo`, so I created #10824 to fix it.

blueyed committed:
-u NONE for no syntax highlighting

chrisbra commented:
> Likely because it's run before installation.

Try explicitly export $VIMRUNTIMEDIR

blueyed commented:
> > Likely because it's run before installation.
> 
> Try explicitly export $VIMRUNTIMEDIR

Good idea in general.

For now `-u NONE` did it: https://travis-ci.org/neovim/neovim/jobs/574710566#L6552
(Neovim enabled syntax highlighting automatically with any config)

blueyed commented:
Let's rebase this after / wait for https://github.com/neovim/neovim/pull/10824 then.

erw7 commented:
@blueyed Can you put `-u NONE` in `NO_INITS` if it doesn't cause any problems in `NO_INITS`? Even with `fixff`, an error like `E484: Can't open file C:/Program Files (x86)/nvim/share/nvim\syntax\syntax.vim` appears.

blueyed commented:
@erw7 
> @blueyed Can you put `-u NONE` in `NO_INITS` if it doesn't cause any problems in `NO_INITS`? Even with `fixff`, an error like `E484: Can't open file C:/Program Files (x86)/nvim/share/nvim\syntax\syntax.vim` appears.

Thought about this also already, and it might be good to avoid problems.

But OTOH `-u` is meant to be used always explicitly, and this shows where it is missing then.

I'd rather have the "-u" option not duplicated unnecessarily.

This will be reworked/revisited with #10651 then anyway I assume.

blueyed commented:
@erw7 
> Even with `fixff`, an error like `E484: Can't open file C:/Program Files (x86)/nvim/share/nvim\syntax\syntax.vim` appears.

To clarify: `-u NONE` would not help there, since `-u unix.vim` is used then.
Having `-i NONE` in `NO_INITS` should have fixed it by now, right?
